system/firewalld.py

```diff
@@ -112,13 +112,197 @@
 # Globals
 #
 fw = None
+msgs = None
 module = None
-fw_offline = False
+changed = False
+immediate = None
+permanent = None
 Rich_Rule = None
+fw_offline = False
+desired_state = None
 FirewallClientZoneSettings = None
 
 module = None
 
+
+#####################
+# action logic
+#
+def action_runner(
+    action,
+    get_enabled_immediate, get_enabled_immediate_args,
+    get_enabled_permanent, get_enabled_permanent_args,
+    set_enabled_immediate, set_enabled_immediate_args,
+    set_enabled_permanent, set_enabled_permanent_args,
+    set_disabled_immediate, set_disabled_immediate_args,
+    set_disabled_permanent, set_disabled_permanent_args
```

Comment:
Or just pass in msgs_tuple as one extra parameter to this function... Have to look at the callers to see which would work better.